File: CCC '03 S1 - Snakes and Ladders.py
running = True
starting_square = 1
while running:
    diceNum = int(input("Enter number: "))

    if diceNum == 0:
        print("You Quit!")
        running = False

    starting_square += diceNum

    if starting_square < 101:

        if starting_square == 54:
            starting_square = 19
            print("You are on square " + str(starting_square))
        elif starting_square == 90:
            starting_square = 48
            print("You are on square " + str(starting_square))
        elif starting_square == 99:
            starting_square = 77
            print("You are on square " + str(starting_square))
        elif starting_square == 9:
            starting_square = 34
            print("You are on square " + str(starting_square))
        elif starting_square == 40:
            starting_square = 64
            print("You are on square " + str(starting_square))
        elif starting_square == 67:
            starting_square = 86
            print("You are on square " + str(starting_square))
        else:
            print("You are on square " + str(starting_square))

        if starting_square == 100:
            print("You Win!")
            running = False
    else:
        starting_square -= diceNum
        print("You are on square " + str(starting_square))

File: CCC/2003/ CCC '03 S1 - Snakes and Ladders .cpp
#include <bits/stdc++.h>

using namespace std;

int diceNum;
int position = 1;

int main() {

    while (position != 100) {

        cin >> diceNum;

        if (diceNum == 0) {
            cout << "You Quit!" << endl;
            return 0;
        }

        if (position + diceNum <= 100) {
            position += diceNum;
        }

        if (position == 99) {
            position = 77;
        } else if (position == 90) {
            position = 48;
        } else if (position == 54) {
            position = 19;
        } else if (position == 9) {
            position = 34;
        } else if (position == 40) {
            position = 64;
        } else if (position == 67) {
            position = 86;
        }

        cout << "You are now on square " << position << endl;
    }

    cout << "You Win!" << endl;
    return 0;
}